tests/unittests/serving/test_experiments_endpoint.py

```diff
@@ -0,0 +1,209 @@
+              'datetime': datetime.datetime(1, 1, 1),
+              'VCS': {"type": "git",
+                      "is_dirty": False,
+                      "HEAD_sha": "test",
+                      "active_branch": None,
+                      "diff_sha": "diff"}},
+    version=1,
+    pool_size=1,
+    max_trials=10,
+    working_dir='',
+    algorithms={'random': {'seed': 1}},
+    producer={'strategy': 'NoParallelStrategy'},
+)
+
+base_trial = {
+    'experiment': None,
+    'status': 'new',
+    'worker': None,
+    'submit_time': datetime.datetime(1, 1, 1),
+    'start_time': datetime.datetime(1, 1, 1, second=10),
+    'end_time': datetime.datetime(1, 1, 2),
+    'heartbeat': None,
+    'results': [
+        {'name': 'loss', 'type': 'objective', 'value': 0.05},
+        {'name': 'a', 'type': 'statistic', 'value': 10},
+        {'name': 'b', 'type': 'statistic', 'value': 5},
+    ],
+    'params': [
+        {'name': 'x',
+         'type': 'real',
+         'value': 10.0},
+    ],
+}
+
+
+class TestCollection:
+    """Tests the server's response on experiments/"""
+
+    def test_no_experiments(self, client):
+        """Tests that the API returns a positive response when no experiments are present"""
+        response = client.simulate_get('/experiments')
+
+        assert response.json == []
+        assert response.status == "200 OK"
+
+    def test_send_name_and_versions(self, client):
+        """Tests that the API returns all the experiments with their name and version"""
+        expected = [
+            {'name': 'a', 'version': 1},
+            {'name': 'b', 'version': 1}
+        ]
+
+        _add_experiment(name='a', version=1, _id=1)
+        _add_experiment(name='b', version=1, _id=2)
+
+        response = client.simulate_get('/experiments')
+
+        assert response.json == expected
+        assert response.status == "200 OK"
+
+    def test_latest_versions(self, client):
+        """Tests that the API return the latest versions of each experiment"""
+        expected = [
+            {'name': 'a', 'version': 2},
+            {'name': 'b', 'version': 1}
+        ]
+
+        _add_experiment(name='a', version=1, _id=1)
+        _add_experiment(name='a', version=2, _id=2)
+        _add_experiment(name='b', version=1, _id=3)
+
+        response = client.simulate_get('/experiments')
+
+        assert response.json == expected
+        assert response.status == "200 OK"
+
+
+class TestItem:
+    """Tests the server's response to experiments/:name"""
+
+    def test_non_existent_experiment(self, client):
+        """
+        Tests that a 404 response is returned when the experiment
+        doesn't exist in the database
+        """
+        response = client.simulate_get('/experiments/a')
+
+        assert response.status == "404 Not Found"
+        assert response.json['message']
+        assert response.json['message'] == "Experiment 'a' does not exist"
+
+        _add_experiment(name='a', version=1, _id=1)
```

Comment:
When add a and test for b? You want to test when there is at least one exp in the DB to make sure the request would fail properly?

Reply:
Exactly. I'm testing against an empty database and on a "full" database to make sure the behavior is consistent.